x-pack/plugins/maps/public/connected_components/timeslider/timeslider.tsx

```diff
@@ -7,248 +7,105 @@
+import {
+  ControlGroupContainer,
+  type ControlGroupInput,
+  type controlGroupInputBuilder,
+  LazyControlGroupRenderer,
+} from '@kbn/controls-plugin/public';
+import { withSuspense } from '@kbn/presentation-util-plugin/public';
 import { first } from 'rxjs/operators';
 import type { TimeRange } from '@kbn/es-query';
-import { epochToKbnDateFormat, getInterval, getTicks } from './time_utils';
-import { getTimeFilter } from '../../kibana_services';
 import { Timeslice } from '../../../common/descriptor_types';
 
+const ControlGroupRenderer = withSuspense(LazyControlGroupRenderer);
+
 export interface Props {
-  closeTimeslider: () => void;
-  setTimeslice: (timeslice: Timeslice) => void;
-  isTimesliderOpen: boolean;
+  setTimeslice: (timeslice?: Timeslice) => void;
   timeRange: TimeRange;
   waitForTimesliceToLoad$: Observable<void>;
-  updateGlobalTimeRange: (timeslice: number[]) => void;
-}
-
-interface State {
-  isPaused: boolean;
-  max: number;
-  min: number;
-  range: number;
-  timeslice: [number, number];
-  ticks: EuiRangeTick[];
-}
-
-function prettyPrintTimeslice(timeslice: [number, number]) {
-  return `${epochToKbnDateFormat(timeslice[0])} - ${epochToKbnDateFormat(timeslice[1])}`;
 }
 
-// Why Timeslider and KeyedTimeslider?
-// Using react 'key' property to ensure new KeyedTimeslider instance whenever props.timeRange changes
-export function Timeslider(props: Props) {
-  return props.isTimesliderOpen ? (
-    <KeyedTimeslider key={`${props.timeRange.from}-${props.timeRange.to}`} {...props} />
-  ) : null;
-}
-
-class KeyedTimeslider extends Component<Props, State> {
+export class Timeslider extends Component<Props, {}> {
   private _isMounted: boolean = false;
-  private _timeoutId: number | undefined;
-  private _subscription: Subscription | undefined;
-
-  constructor(props: Props) {
-    super(props);
-    const timeRangeBounds = getTimeFilter().calculateBounds(props.timeRange);
-    if (timeRangeBounds.min === undefined || timeRangeBounds.max === undefined) {
-      throw new Error(
-        'Unable to create Timeslider component, timeRangeBounds min or max are undefined'
-      );
-    }
-    const min = timeRangeBounds.min.valueOf();
-    const max = timeRangeBounds.max.valueOf();
-    const interval = getInterval(min, max);
-    const timeslice: [number, number] = [min, max];
-
-    this.state = {
-      isPaused: true,
-      max,
-      min,
-      range: interval,
-      ticks: getTicks(min, max, interval),
-      timeslice,
-    };
-  }
+  private _controlGroup?: ControlGroupContainer | undefined;
+  private readonly _subscriptions = new Subscription();
 
   componentWillUnmount() {
-    this._onPause();
     this._isMounted = false;
+    this._subscriptions.unsubscribe();
   }
 
-  componentDidMount() {
-    this._isMounted = true;
-    // auto-select range between first tick and second tick
-    this._onChange([this.state.ticks[0].value, this.state.ticks[1].value]);
+  componentDidUpdate() {
+    if (
```

Comment:
Eventually this logic is something that the control group itself could handle, potentially even by handing it a subject on mount. This is looking good so far though!